app/services/dttp/retrieve_account.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Dttp
+  class RetrieveAccount
+    include ServicePattern
+
+    class HttpError < StandardError; end
+
+    attr_reader :provider
+
+    ACCOUNT_ATTRIBUTE_FIELDS = %w[
+      name
+    ].freeze
+
+    def initialize(provider:)
+      @provider = provider
+    end
+
+    def call
+      response = Client.get("/accounts(#{provider.dttp_id})?$select=#{fields}")
+      if response.code != 200
+        raise HttpError, "status: #{response.code}, body: #{response.body}, headers: #{response.headers}"
+      end
+
+      JSON(response.body)
```

Comment:
Can we return an object instead of a hash so as we can add accessor methods with consistent naming etc instead of the calling code needing to know the field names. e.g. `Dttp::Account`. We can just expose `Dttp::Account#name` for now.

Reply:
fixed, methods now returns object with accessor methods defined inside. Pattern should now be consistent across all our three PRs